### PR TITLE
Updating test infra role references

### PR DESCRIPTION
**Pull request by claurence**
updating the locations the test infra role is mentioned in the SIG-Release Repo 

/assign @justaugustus @tpepper 

cc @imkin @Bubblemelon

### CODE REVIEW

File: release-team/README.md [hunk 1]
```diff
@@ -76,7 +76,6 @@ there are 3 types of Kubernetes releases:
 | Release Team Lead | [Lead Handbook](role-handbooks/release-team-lead/README.md) |
 | Enhancements | [Enhancements Handbook](role-handbooks/enhancements/README.md) |
 | CI Signal | [CI Signal Handbook](role-handbooks/ci-signal/README.md) |
-| Test Infra | [Test Infra Handbook](role-handbooks/test-infra/README.md) |
```
Comment (imkin) [inline release-team/README.md]: Woot!!!!

File: release-team/role-handbooks/test-infra/README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
-# Test Infra Lead Handbook
+# Test Infra Lead Handbook 
+Note: This role was deprecated after the 1.15 release.  Starting with 1.16 this role was mostly merged with [Branch Management](https://github.com/sig-release/blob/updating-test-infra-role-references/release-team/role-handbooks/branch-manager/README.md)
```
Comment (imkin) [inline release-team/role-handbooks/test-infra/README.md]: The responsibliities were split between branch manager and test-infra-oncall. 
Hence we should indicate that all the maintenance and test-infra related questions should be directed to test-infra-oncall at the channels #testing-ops #sig-testing.

**Comment by justaugustus** [comment]
@claurence -- Also, we should put a note/table under the [current table for handbooks](https://github.com/kubernetes/sig-release/tree/master/release-team#kubernetes-release-team-roles) denoting retired roles (and maybe the cycle they were retired in).

File: release-team/role-handbooks/test-infra/README.md (hunk 2 above)
Comment (Bubblemelon) [reply release-team/role-handbooks/test-infra/README.md]: > ... maintenance and test-infra related questions should be directed to test-infra-oncall at the channels #testing-ops #sig-testing.

Adding this to the branch manager handbook as well.

**Comment by justaugustus** [comment]
I'll issue additional cleanups as follow-up PRs.
/hold cancel
/lgtm
/approve

**Comment by justaugustus** [comment]
Follow-up PR here: #693

**Pull request merged**